pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -423,6 +504,87 @@ pub mod pallet {
 
 			T::InboundQueue::submit(domain_address, incoming_msg)
 		}
+
+		/// Convenience method for manually processing an outbound message.
+		///
+		/// If the execution fails, the message gets moved to the
+		/// `FailedOutboundMessages` storage.
+		#[pallet::weight(T::WeightInfo::process_outbound_message())]
+		#[pallet::call_index(6)]
+		pub fn process_outbound_message(
+			origin: OriginFor<T>,
+			nonce: T::OutboundMessageNonce,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			let (domain, sender, message) = OutboundMessageQueue::<T>::take(nonce)
+				.ok_or(Error::<T>::OutboundMessageNotFound)?;
+
+			match Self::process_message(domain.clone(), sender.clone(), message.clone()) {
+				Ok(_) => {
+					Self::deposit_event(Event::<T>::OutboundMessageExecutionSuccess {
+						nonce,
+						domain,
+						sender,
+						message,
+					});
+
+					Ok(())
+				}
+				Err(e) => {
+					Self::deposit_event(Event::<T>::OutboundMessageExecutionFailure {
+						nonce,
+						domain: domain.clone(),
+						sender: sender.clone(),
+						message: message.clone(),
+						error: e.error,
+					});
+
+					FailedOutboundMessages::<T>::insert(nonce, (domain, sender, message, e.error));
+
+					Ok(())
+				}
+			}
+		}
+
+		/// Manually process a failed outbound message.
+		#[pallet::weight(T::WeightInfo::process_failed_outbound_message())]
+		#[pallet::call_index(7)]
+		pub fn process_failed_outbound_message(
```

Comment:
Note how we always return `Ok(())` here so that we can preserve the storage changes in case of errors.